src/compiler/checker.rs

```diff
@@ -1004,23 +1151,72 @@ impl<'type_checker> CheckTypeRelatedTo<'type_checker> {
         Ternary::False
     }
 
+    fn each_type_related_to_type(
+        &self,
+        source: &UnionOrIntersectionType,
+        target: Rc<Type>,
+        report_errors: bool,
+        intersection_state: IntersectionState,
+    ) -> Ternary {
+        let mut result = Ternary::True;
+        let source_types = source.types();
+        for source_type in source_types {
+            let related = self.is_related_to(
+                source_type.clone(),
+                target.clone(),
+                report_errors,
+                None,
+                Some(intersection_state),
+            );
+            if related == Ternary::False {
+                return Ternary::False;
+            }
+            result &= related;
```

Comment:
The Typescript definition of the `Ternary` enum (where the enum variant values are `0`, `1`, `3` and `-1`) explains how they have those values so that bitwise-`&`'ing/`|`'ing them together gives you the greater or lesser of the two (I don't understand how `-1` works in terms of bitwise operations). I did a little sanity-checking that the same seems to hold true with Rust `i32`'s and then implemented `BitAndAssign` for `Ternary`

Reply:
🤯